codice/ddf
# [2.19.x] DDF-5672 fix dms lat/lon entry UX

#### What does this PR do?
Improves the UX for entering latitude/longitude(DMS) values.

Previously, the input validation would "right-pad" the user input with 0's, so if the user entered "2", it would place the 2 in the far left of the template, causing the value to register as "200°00'00", which is an invalid longitude value. This would cause the value to change to the max "180°00'00".

This PR improves the UX by "left-padding" the input with 0's, for each component, separately. In order to achieve this, the placeholder string was changed from a `0` to a `_`. Then, after the user finishes inputting and the focus changes away from the input field, the `_` placeholders will be removed and the number will be padded appropriately. Note: any numbers after a decimal place will be right-padded with 0's. 

#### Who is reviewing it? 

@zta6 @andrewzimmer 
#### Select relevant component teams: 


#### Ask 2 committers to review/merge the PR and tag them here.


#### How should this be tested?

1. Go to the search UI
2. Create an advanced search with anyGeo, and select either Bounding Box or Point Radius (anything that lets you input DMS-formatted coordinates)
3. Select Lat/Lon (DMS)
4. Start manually entering coordinates into any of the coordinate input fields
5. Verify that entering a value such as a "2" doesn't break the validation
6. Click or tab away from the input field and verify that the coordinate is formatted as expected

#### Any background context you want to provide?

#### What are the relevant tickets?
Fixes: #5672 5672

#### Screenshots


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
